.github/workflows/setup_env.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+#-------------------------------------------------------------------------------
+# Functions
+
+# Suppress echo (-x) output of commands executed with `quiet`.
+# Useful for sourcing files, loading modules, spack, etc.
+# set +x, set -x are not echo'd.
+quiet() {
+    { set +x; } 2> /dev/null;
+    $@;
+    set -x
+}
+
+# `print` is like `echo`, but suppresses output of the command itself.
+# https://superuser.com/a/1141026
+echo_and_restore() {
+    builtin echo "$*"
+    date
+    case "${save_flags}" in
+        (*x*)  set -x
```

Comment:
Stupid question. I usually have a single parenthesis only at the end of the possible cases in the switch. I see one at the beginning. Is it needed?

Reply:
I copied from the above superuser link. Turns out it's optional but makes balanced parenthesis. https://stackoverflow.com/questions/65703886/does-starting-parenthesis-in-bash-case-statement-have-meaning